### PR TITLE
Thread safe improvements

### PR DESCRIPTION
**Pull request by Grigory-Rylov**

### CODE REVIEW

File: src/main/java/com/github/grishberg/tests/planner/InstrumentalTestPlanProvider.java [hunk 1]
```diff
@@ -12,20 +12,21 @@
...
  * Provides set of {@link TestPlanElement} for instrumental tests.
+ * This class is thread safe.
  */
 public class InstrumentalTestPlanProvider {
     private static final String TAG = InstrumentalTestPlanProvider.class.getSimpleName();
     private final InstrumentalExtension instrumentationInfo;
     private final Map<String, String> propertiesMap;
     private final PackageTreeGenerator packageTreeGenerator;
-    private RunnerLogger logger;
+    private final RunnerLogger logger;
```
Comment (Nexen23) [inline src/main/java/com/github/grishberg/tests/planner/InstrumentalTestPlanProvider.java]: Logger itself can be not thread-safe. As well as all packageTreeGenerator and other.
If you create this class on one thread and then use it on another, seems like `provideTestPlan()` won't be thread-safe completely.

File: src/main/java/com/github/grishberg/tests/planner/PackageTreeGenerator.java [hunk 2]
```diff
@@ -8,6 +8,7 @@
 
 /**
  * Generates tree of classes and packages.
+ * This class is thread safe.
```
Comment (Nexen23) [inline src/main/java/com/github/grishberg/tests/planner/PackageTreeGenerator.java]: It seems it's better to call it stateless

File: src/main/java/com/github/grishberg/tests/ConnectedDeviceWrapper.java [hunk 3]
```diff
@@ -55,21 +55,10 @@ public int getDensity() {
...
-        } catch (CommandExecutionException e) {
-            logger.e(TAG, "calculateScreenSize error: ", e);
-        }
-    }
-
     /**
      * @return device screen width.
      */
-    public int getWidth() {
+    public synchronized int getWidth() {
```
Comment (Nexen23) [inline src/main/java/com/github/grishberg/tests/ConnectedDeviceWrapper.java]: You're protecting local methods, but what about `IDevice` itself? Is it actually thread-safe too?

File: src/main/java/com/github/grishberg/tests/planner/InstrumentalTestPlanProvider.java (hunk 1 above)
Comment (Grigory-Rylov) [reply src/main/java/com/github/grishberg/tests/planner/InstrumentalTestPlanProvider.java]: `PackageTreeGenerator` is thread-safe
there is thread-safe copy of `InstrumentalExtension`
there is cope of `propertiesMap`.
`RunnerLogger` does not affect the logic of this class
That is why I think `InstrumentalTestPlanProvider` something like thread-safe

File: src/main/java/com/github/grishberg/tests/planner/PackageTreeGenerator.java (hunk 2 above)
Comment (Grigory-Rylov) [reply src/main/java/com/github/grishberg/tests/planner/PackageTreeGenerator.java]: Why? It is thread-safe class) isn't it?

File: src/main/java/com/github/grishberg/tests/ConnectedDeviceWrapper.java (hunk 3 above)
Comment (Grigory-Rylov) [reply src/main/java/com/github/grishberg/tests/ConnectedDeviceWrapper.java]: You are right, need to make every calls to IDevice synchronised

File: src/main/java/com/github/grishberg/tests/planner/InstrumentalTestPlanProvider.java (hunk 1 above)
Comment (Nexen23) [reply src/main/java/com/github/grishberg/tests/planner/InstrumentalTestPlanProvider.java]: `propertiesMap` - seems ctor  of HashMap makes it not thread-safe. Better use syncronizedMap

Comment (Nexen23) [reply src/main/java/com/github/grishberg/tests/planner/InstrumentalTestPlanProvider.java]: Same problem with InstrumentalExtension

File: src/main/java/com/github/grishberg/tests/planner/PackageTreeGenerator.java (hunk 2 above)
Comment (Grigory-Rylov) [reply src/main/java/com/github/grishberg/tests/planner/PackageTreeGenerator.java]: this.propertiesMap = Collections.synchronizedMap(propertiesMap);

**Comment by Grigory-Rylov** [comment]
@Nexen23 about final collection and thread safety  when only readings : https://stackoverflow.com/questions/6457109/java-concurrency-is-final-field-initialized-in-constructor-thread-safe

I made map unmodifiableMap, I think this will be enough.
Please, review this PR

**Comment by Grigory-Rylov** [comment]
```
The ability to see the correctly constructed value for the field is nice, but if the field itself is a reference, then you also want your code to see the up to date values for the object (or array) to which it points. If your field is a final field, this is also guaranteed. So, you can have a final pointer to an array and not have to worry about other threads seeing the correct values for the array reference, but incorrect values for the contents of the array. Again, by "correct" here, we mean "up to date as of the end of the object's constructor", not "the latest value available". 
```

**Comment by Nexen23** [comment]
> 
> 
> The ability to see the correctly constructed value for the field is nice, but if the field itself is a reference, then you also want your code to see the up to date values for the object (or array) to which it points. If your field is a final field, this is also guaranteed. So, you can have a final pointer to an array and not have to worry about other threads seeing the correct values for the array reference, but incorrect values for the contents of the array. Again, by "correct" here, we mean "up to date as of the end of the object's constructor", not "the latest value available".

It would be better if you pasted a link here which you have given me with quote from that docs : )

**Pull request merged**